src/colmap/scene/reconstruction.cc

```diff
@@ -359,6 +359,18 @@ void Reconstruction::AddImageWithTrivialFrame(class Image image,
 void Reconstruction::AddPoint3D(const point3D_t point3D_id,
                                 struct Point3D point3D) {
   max_point3D_id_ = std::max(max_point3D_id_, point3D_id);
+
+  for (const auto& track_el : point3D.track.Elements()) {
+    class Image& image = Image(track_el.image_id);
+    const Point2D& point2D = image.Point2D(track_el.point2D_idx);
+    if (point2D.HasPoint3D()) {
+      THROW_CHECK_EQ(point2D.point3D_id, point3D_id);
+    } else {
+      image.SetPoint3DForPoint2D(track_el.point2D_idx, point3D_id);
+    }
```

Comment:
Should we do the same for the other AddPoint3D method below or ideally even consolidate the two? I guess the bug was introduced due to not realizing that there are two parallel methods almost doing the same thing?